core/src/main/java/com/scalar/db/io/Value.java

```diff
@@ -1,13 +1,14 @@
 package com.scalar.db.io;
 
 import java.util.Optional;
+import javax.annotation.Nonnull;
 
 /**
  * An abstraction for storage entry's value
  *
  * @author Hiroyuki Yamada
  */
-public interface Value<T> extends Comparable<T> {
+public interface Value<T> extends Comparable<Value<T>> {
```

Comment:
I think this is the biggest change. Please take a look at this carefully.

Reply:
I'm not fully sure, but should we implement `Comparable` in each sub-class so that we can have, for example, `compareTo(BigIntValue)` in BigIntValue and we can avoid runtime casting causing ClassCastException?

Reply:
Thank you for taking a look at this!

Regarding this, Error Prone showed the following warning:
https://errorprone.info/bugpattern/ComparableType

Reply:
I meant, I think it seems better to remove the `extends Comparable<Value<T>>` from Value class and add type-specific `implements Comparable` to each subclass like `implements Comparable<BitIntValue>` in BigIntValue class from users' perspective.

Reply:
Ah, that can be a good idea. Let me try. Thanks.